doc/storage/api/api.rst

```diff
@@ -350,10 +509,29 @@ These definitions must be defined in the header file :file:`psa/protected_storag
 
     It will be incremented in small updates that are unlikely to include breaking changes.
 
+.. macro:: PSA_PS_SUPPORT_ITERATION
+    (1u << 2)
+
+    Flag indicating that `psa_ps_iterator_start` and `psa_ps_iterator_next` are supported.
+
+.. struct:: psa_ps_storage_iterator_t
+
+    .. summary::
+        An implementation-defined opaque structure containing the context for an iterator.
+        The structure MUST contain all all the state required by the iterator.
+        That is, further state MUST NOT be retained by the implementation.
+
+        The structure is initilaised by the ``ps_iterator_start()`` function.
+        It is modified by the ``ps_iterator_next()`` function.
+
+        the caller can discard or reuse the iterator object once it has finished using it. This can be before, or after, the iterator has reached the end of the iteration.
```

Comment:
```suggestion
        The caller can discard or reuse the iterator object once it has finished using it. This can be before, or after, the iterator has reached the end of the iteration.
```